### DISCUSSION TITLE
Azure B2C Authentication in new .net 8 blazor auto render mode

### DISCUSSION THREAD

**Discussion by satyajit-behera** [dotnet/aspnetcore#53631] [Q&A]
Hi,
The new blazor 8 web app auto mode has both server and client side rendering enabled.
Is there any sample and documentation that explains implementing Azure AD B2C while creating web application using Auto render mode.
Thanks

**Comment by halter73** [comment]
We're currently working on a sample for this. You can check it out at https://github.com/dotnet/blazor-samples/tree/ea382520bf99b195b7c64108ca0bcc7e9ea49ca5/8.0/BlazorWebAppOidc and https://github.com/dotnet/blazor-samples/tree/ea382520bf99b195b7c64108ca0bcc7e9ea49ca5/8.0/BlazorWebAppOidcBff.

There's an PR open at https://github.com/dotnet/AspNetCore.Docs/pull/31555 for the documentation to go with the sample. You can see the documentation in its current form at https://github.com/dotnet/AspNetCore.Docs/blob/602fb15bba8063cfefb691f099f9cd8dda7dedd0/aspnetcore/blazor/security/server/blazor-web-app-with-oidc.md.

**Reply by nithin-talentzq** [reply]
@halter73  how to invoke external  API's from blazor client side with access token, instead of proxying through server, our backend completely built on azure functions , these functions needs to invoke from blazor client side with access token
